### DISCUSSION TITLE
Generate non-HTML files on `next export`

### DISCUSSION THREAD

**Discussion by r7kamura** [vercel/next.js#36640] [Ideas]
### Describe the feature you'd like to request

When using `next export`, I would like to generate non-HTML files, but this is not currently supported.

For example, we may want to do the following.

- `GET /feed.xml` to return XML for RSS
- `GET /sitemap.txt` to return text for Google crawler

With SSR, we can use API Routes and Rewrites, or `getServerProps` to prepare such endpoints, but with SSG by `next export`, none of these functions are available.

Are there any plans to support the ability to generate non-HTML files with `next export`?

### Describe the solution you'd like

If we can default export functions that return non-HTML content on regular pages, we should be able to achieve this using `getStaticPaths` and `getStaticProps`.

### Describe alternatives you've considered

I saw the following workaround in some blog posts:

- Create non-HTML files in `./public` directory between `next build` and `next export`
    - e.g. https://github.com/iamvishnusankar/next-sitemap recommends this apporach
- Create non-HTML files in `./public` directory in `getStaticProps` of some pages

**Comment by artginzburg** [comment]
This would be awesome. I'd use it to generate `public/manifest.json` for PWAs in a scriptable way. And the `favicon.ico` plus png logos in x192 and x512 generated from an SVG logo.

Generating sitemaps and `robots.txt` is also useful, but they're already almost fine with the postbuild approach since the website's visible behavior itself doesn't depend on them.

We need to be able to generate non-HTML content within the Next.js build system (fast refresh and all that). I imagine it wouldn't be too hard to implement — the same HTML, but without the `<head>` / the scripts / injecting anything complicated  — just plain text. I'd be happy being able to output plain text and specify an arbitrary file extension. Nothing more is really necessary.
I'd generally even opt out of the pre-defined `public` folder, since it's completely static, and for the most part, it's the same copy-pasted structure in every project, with a slight change.

The most simplistic alternative I'm using for now — which is to have `.json` files and importing them into TypeScript — hurts the order of dependencies (forces high-level to depend on low-level). And obviously, that only works acceptably well for key-value/tree structured files.
